lib/generators/boring/devise_jwt/install/install_generator.rb

```diff
@@ -0,0 +1,200 @@
+# frozen_string_literal: true
+
+module Boring
+  module DeviseJwt
+    class InstallGenerator < Rails::Generators::Base
+      desc "add devise-jwt to the application"
+
+      class_option :model_name, type: :string, aliases: "-m",
+                   default: "User",
+                   desc: "Tell us the user model name which will be used for authentication. Defaults to User"
+      class_option :use_rails_credentials, type: :boolean, aliases: "-rc",
+                   desc: "Use Rails credentials for devise_jwt_secret_key"
+      class_option :revocation_strategy, type: :string, aliases: "-rs",
+                   enum: %w[JTIMatcher Denylist Allowlist],
+                   default: "Denylist",
+                   desc: "Tell us the revocation strategy to be used. Defaults to Denylist"
+
+      def verify_presence_of_devise_gem
+        gem_file_content_array = File.readlines("Gemfile")
+        devise_is_installed = gem_file_content_array.any? { |line| line.include?('devise') }
+
+        return if devise_is_installed
+
+        say "We couldn't find devise gem. Please configure devise gem and run the generator again!", :red
+
+        abort
+      end
+
+      def verify_presence_of_devise_initializer
+        return if File.exist?("config/initializers/devise.rb")
+
+        say "We couldn't find devise initializer. Please configure devise gem and run the generator again!", :red
+
+        abort
+      end
+
+      def verify_presence_of_devise_model
+        return if File.exist?("app/models/#{options[:model_name].underscore}.rb")
+
+        say "We couldn't find the #{options[:model_name]} model. Please provide correct model name", :red
+
+        abort
+      end
+
+      def add_devise_jwt_gem
+        say "Adding devise-jwt gem", :green
+        Bundler.with_unbundled_env do
+          run "bundle add devise-jwt"
+        end
+      end
+
+      def add_devise_jwt_config_to_devise_initializer
+        say "Adding devise-jwt config to devise initializer", :green
+        jwt_config = <<~RUBY
+          config.jwt do |jwt|
+            jwt.secret = #{devise_jwt_secret_key}
+            jwt.dispatch_requests = [
+              ['POST', %r{^/sign_in$}]
+            ]
+            jwt.revocation_requests = [
+              ['DELETE', %r{^/sign_out$}]
+            ]
+            jwt.expiration_time = 15.day.to_i
```

Comment:
Can default to 15 days but let's add an option to configure this as well

Reply:
will it okay to allow them to provide expirations time only day unit

Reply:
Yes let's do that, class option name should be clear that it's only accepting days